.github/workflows/standalone-fleet-multicluster-ci.yml

```diff
@@ -0,0 +1,179 @@
+      -
+        name: Set Up k3d Cluster
+        uses: AbsaOSS/k3d-action@v2
+        with:
+          cluster-name: "upstream"
+          args: >-
+            -p "80:80@agent:0:direct"
+            -p "443:443@agent:0:direct"
+            --api-port 6443
+            --agents 3
+            --network "nw01"
+      -
+        name: Set Up k3d Downstream Cluster
+        uses: AbsaOSS/k3d-action@v2
+        with:
+          cluster-name: "downstream"
+          args: >-
+            -p "81:80@agent:0:direct"
+            -p "444:443@agent:0:direct"
+            --api-port 6644
+            --agents 1
+            --network "nw01"
+      -
+        name: K3d Import Images
+        run: |
+          k3d image import rancher/fleet:dev rancher/fleet-agent:dev -c upstream
+          k3d image import rancher/fleet-agent:dev -c downstream
+      -
+        name: Set Up Tmate Debug Session
+        if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.enable_tmate }}
+        uses: mxschmitt/action-tmate@v3
+        timeout-minutes: 15
+        with:
+          limit-access-to-actor: true
+      -
+        name: Deploy Fleet
+        run: |
+          kubectl config use-context k3d-upstream
+          helm -n fleet-system install --create-namespace --wait fleet-crd charts/fleet-crd
+          helm -n fleet-system upgrade --install --create-namespace --wait fleet charts/fleet
+          kubectl -n fleet-system rollout status deploy/fleet-controller
+          { grep -q -m 1 "fleet-agent"; kill $!; } < <(kubectl get deployment -n fleet-system -w)
+          kubectl -n fleet-system rollout status deploy/fleet-agent
+      -
+        name: Deploy and Register Downstream Fleet
+        run: |
+          kubectl apply -f - <<EOF
+          apiVersion: "fleet.cattle.io/v1alpha1"
+          kind: ClusterRegistrationToken
+          metadata:
+            name: second-token
+            namespace: fleet-local
+          spec:
+            ttl: 12h
+          EOF
+          { grep -q -m 1 "second-token"; kill $!; } < <(kubectl get secrets -n fleet-local -w)
+
+          token=$(kubectl get secret -n fleet-local second-token -o go-template='{{index .data "values" | base64decode}}' | yq eval .token -)
+          name=$(kubectl get -n default sa default -o=jsonpath='{.secrets[0].name}')
+          ca=$(kubectl get -n default secret "$name" -o go-template='{{index .data "ca.crt" | base64decode}}')
+
+          kubectl config use-context k3d-downstream
+          helm -n fleet-system upgrade --install --create-namespace --wait fleet-agent charts/fleet-agent \
+            --set-string labels.env=dev \
+            --set apiServerCA="$ca" \
+            --set apiServerURL="https://172.18.0.1.omg.howdoi.website:6443" \
+            --set clusterNamespace="fleet-local" \
+            --set systemRegistrationNamespace="fleet-clusters-system" \
+            --set token="$token"
```

Comment:
I feel like non-trivial scripts like this should be placed into a common set of scripts under `./scripts`; this will also prevent us from duplicating the same script logic in both workflows.

Perhaps we should make these bash scripts under `./scripts/bootstrap/*.sh`?

Reply:
On the other hand the script is trivial, no parameters, no forks, as long as it has one purpose and is not in the common script folder.
Duplication is less maintenance than the wrong abstraction. 

I previously put the "setup-rancher" scripts into `.github/assets`, because I didn't want to suggest that these scripts should be used by anyone but the workflows.
I do not intend to build re-usable scripts. I don’t want to make the journey easier for devs and their local test setups (not in this PR I mean ;) ). I want to avoid parameters and conditionals.

So that’s why the workflows are mostly self-contained, have one user / environment (github actions runner) and don’t use shared scripts. As long as we don’t have re-use in maybe three places, I’m happy with copy and paste. We might decide to use terraform instead and all of this can be deleted soon. 

I have https://github.com/manno/fleet-dev-tools/tree/main/bin and use that myself for local testing, but I'm not ready to turn these into general-purpose scripts and maintain them in the main repo.